partiql-conformance-tests/build.rs

```diff
@@ -0,0 +1,296 @@
+    let ion_file_iterator = WalkDir::new(dir)
+        .into_iter()
+        .map(|entry| {
+            entry.unwrap_or_else(|error| panic!("Failure during dir traversal: {:?}", error))
+        })
+        .filter(|entry| entry.path().is_file())
+        .map(|entry| entry.path().to_owned())
+        .filter(|f| f.extension() == Some(ion_file_extension));
+    ion_file_iterator.collect()
+}
+
+/// Returns a file name with prefix prepended and 'rs' as the extension
+fn to_full_file_name(prefix: &str, ion_file: &Path) -> String {
+    format!(
+        "{}_{}",
+        prefix,
+        ion_file
+            .with_extension("rs")
+            .file_name()
+            .unwrap()
+            .to_str()
+            .unwrap()
+    )
+}
+
+/// Converts the scope to a String that is passed through codegen's default formatter
+fn scope_to_formatted_string(scope: Scope) -> String {
+    let mut dest_str = String::new();
+    scope
+        .fmt(&mut Formatter::new(&mut dest_str))
+        .unwrap_or_else(|error| panic!("Failure during scope formatting: {:?}", error));
+    dest_str
+}
+
+trait TestFunction {
+    fn to_test_function(&self) -> Function;
+}
+
+/// Represents a conformance test with a PartiQL statement that will parse without errors
+/// TODO: add additional parse checks (e.g. ast)
+struct ParsePassTestCase {
+    test_name: String,
+    statement: String,
+    namespace: Option<String>,
+}
+
+impl TestFunction for ParsePassTestCase {
+    /// Creates a [`codegen::Function`] that checks the parsing of `statement` does not result in
+    /// a parsing error.
+    fn to_test_function(&self) -> Function {
+        let complete_test_name = match &self.namespace {
+            Some(ns) => format!("{}_{}", ns, &self.test_name),
+            None => self.test_name.to_owned(),
+        };
+        let mut generated_fun = Function::new(complete_test_name.replace(' ', "_").as_str());
+        generated_fun
+            .attr("test")
+            .line(format!(
+                "let parse_result = lalr_parse(\"{}\");",
+                &self.statement
+            ))
+            .line("assert!(parse_result.is_ok());");
+        generated_fun
+    }
+}
+
+/// Represents a conformance test with a PartiQL statement that will give an error when parsed
+/// TODO: add additional fields for further error-checking (e.g. error code)
+struct ParseFailTestCase {
+    test_name: String,
+    statement: String,
+    namespace: Option<String>,
+}
+
+impl TestFunction for ParseFailTestCase {
+    /// Creates a [`codegen::Function`] that checks the parsing of `statement` results in a parsing
+    /// error.
+    fn to_test_function(&self) -> Function {
+        let complete_test_name = match &self.namespace {
+            Some(ns) => format!("{}_{}", ns, &self.test_name),
+            None => self.test_name.to_owned(),
+        };
+        let mut generated_fun = Function::new(complete_test_name.replace(' ', "_").as_str());
```

Comment:
`replace(' ', "_")` appears a couple times in this file.

You'll probably want to extract out some `escape_*` functions to do this escaping, as it's not just spaces in the test names that you'll have to deal with eventually (ion allows some punctuation in symbol names and also, you may have to mangle names that happen to also be rust reserved words)

Reply:
Agreeing with Josh; on the other side, if it's too much of an effort, maybe we can just restrict the test names for now.